### PR TITLE
Fix CancellationTokenSource.CancelAfter(TimeSpan delay) ArgumentException documentation

### PR DESCRIPTION
**Pull request by arunjvs**
## Summary

Clarify supported values of TimeSpan delay paramater.

Partly addresses dotnet/runtime#68841

### CODE REVIEW

File: xml/System.Threading/CancellationTokenSource.xml [hunk 1]
```diff
@@ -515,16 +515,19 @@
...
-            
-            Subsequent calls to `CancelAfter` will reset the delay for this <xref:System.Threading.CancellationTokenSource>, if it has not been canceled already.
+
+## Remarks
+ The countdown for the delay starts during this call. When the delay expires, this <xref:System.Threading.CancellationTokenSource> is canceled, if it has not been canceled already.
+
+ Subsequent calls to `CancelAfter` will reset the delay for this <xref:System.Threading.CancellationTokenSource>, if it has not been canceled already.  
+  
+> [!NOTE]
+>  Valid `TimeSpan` objects can be created with a larger range of values than accepted by this method. Calling `CancelAsync` with `TimeSpan delay` values greater than `TimeSpan.FromMilliseconds(Int32.MaxValue)` (or `TimeSpan.FromMilliseconds(UInt32.MaxValue - 1)` on some versions of .NET) upto `TimeSpan.MaxValue` will throw `System.ArgumentOutOfRangeException`.
```
Comment (kouvel) [inline xml/System.Threading/CancellationTokenSource.xml:525]:
```suggestion
>  Valid `TimeSpan` objects can be created with a larger range of values than accepted by this method. Calling `CancelAsync` with `TimeSpan delay` values greater than `TimeSpan.FromMilliseconds(Int32.MaxValue)` (or `TimeSpan.FromMilliseconds(UInt32.MaxValue - 1)` on some versions of .NET) up to `TimeSpan.MaxValue` will throw `System.ArgumentOutOfRangeException`.
```

**Review by kouvel — APPROVED** [review]
Just a minor suggestion, otherwise LGTM, thanks!

File: xml/System.Threading/CancellationTokenSource.xml [hunk 2]
```diff
@@ -515,16 +515,19 @@
...
-            
-            Subsequent calls to `CancelAfter` will reset the delay for this <xref:System.Threading.CancellationTokenSource>, if it has not been canceled already.
+
+## Remarks
+ The countdown for the delay starts during this call. When the delay expires, this <xref:System.Threading.CancellationTokenSource> is canceled, if it has not been canceled already.
+
+ Subsequent calls to `CancelAfter` will reset the delay for this <xref:System.Threading.CancellationTokenSource>, if it has not been canceled already.  
+  
+> [!NOTE]
+>  Valid `TimeSpan` objects can be created with a larger range of values than accepted by this method. Calling `CancelAsync` with `TimeSpan delay` values greater than `TimeSpan.FromMilliseconds(Int32.MaxValue)` (or `TimeSpan.FromMilliseconds(UInt32.MaxValue - 1)` on some versions of .NET) up to `TimeSpan.MaxValue` will throw `System.ArgumentOutOfRangeException`.
```
Comment (gewarren) [inline xml/System.Threading/CancellationTokenSource.xml:525]:
```suggestion
> It's possible to create valid `TimeSpan` objects that have a larger range of values than is accepted by this method. If you call `CancelAfter` with a `TimeSpan` value greater than `TimeSpan.FromMilliseconds(Int32.MaxValue)` (or `TimeSpan.FromMilliseconds(UInt32.MaxValue - 1)` on some versions of .NET) up to `TimeSpan.MaxValue`, an `System.ArgumentOutOfRangeException` is thrown.
```

**Review by gewarren — COMMENTED** [review]
I left a couple suggestions.

File: xml/System.Threading/CancellationTokenSource.xml (hunk 2 above)
Comment (arunjvs) [reply xml/System.Threading/CancellationTokenSource.xml]: Actually this has become too wordy overall, simplified it. The focus in the later part should be on the specific parameter `delay` than the type, which is already know in params section.

File: xml/System.Threading/CancellationTokenSource.xml [hunk 3]
```diff
@@ -515,16 +515,19 @@
...
-            
-            Subsequent calls to `CancelAfter` will reset the delay for this <xref:System.Threading.CancellationTokenSource>, if it has not been canceled already.
+
+## Remarks
+ The countdown for the delay starts during this call. When the delay expires, this <xref:System.Threading.CancellationTokenSource> is canceled, if it has not been canceled already.
+
+ Subsequent calls to `CancelAfter` will reset the delay for this <xref:System.Threading.CancellationTokenSource>, if it has not been canceled already.  
+  
+> [!NOTE]
+>  The `TimeSpan` struct supports a larger range of time intervals than is accepted by this method. `CancelAsync` throws `System.ArgumentOutOfRangeException` if the `delay` parameter is greater than `TimeSpan.FromMilliseconds(Int32.MaxValue)` (or `TimeSpan.FromMilliseconds(UInt32.MaxValue - 1)` on some versions of .NET) up to and including `TimeSpan.MaxValue`.
```
Comment (gewarren) [inline xml/System.Threading/CancellationTokenSource.xml:525]: I had changed this to `CancelAfter`. Did you mean to put `CancelAsync`?

Comment (arunjvs) [reply xml/System.Threading/CancellationTokenSource.xml:525]: Naah my bad, give me a moment, I thinking of dumping the remarks section entirely.

Comment (arunjvs) [reply xml/System.Threading/CancellationTokenSource.xml]: Removed the note and update the constructor also.

**Comment by arunjvs** [comment]
Using same exception language in constructor as well. Removed note.

File: xml/System.Threading/CancellationTokenSource.xml [hunk 4]
```diff
@@ -259,7 +259,8 @@
  ]]></format>
         </remarks>
         <exception cref="T:System.ArgumentOutOfRangeException">
-          <paramref name="delay" />.<see cref="P:System.TimeSpan.TotalMilliseconds" /> is less than -1 or greater than <see cref="F:System.Int32.MaxValue" />.</exception>
+          <paramref name="delay" />.<see cref="P:System.TimeSpan.TotalMilliseconds" /> is less than -1 or greater than Int32.<see cref="F:System.Int32.MaxValue" /> (or UInt32.<see cref="F:System.UInt32.MaxValue" />-1 on some versions of .NET). Note that this upper bound is more restrictive than TimeSpan.<see cref="F:System.TimeSpan.MaxValue" />.
```
Comment (gewarren) [inline xml/System.Threading/CancellationTokenSource.xml:262]:
```suggestion
          <paramref name="delay" />.<see cref="P:System.TimeSpan.TotalMilliseconds" /> is less than -1 or greater than <see cref="F:System.Int32.MaxValue">Int32.MaxValue</see> (or <see cref="F:System.UInt32.MaxValue">UInt32.MaxValue</see> - 1 on some versions of .NET). Note that this upper bound is more restrictive than <see cref="F:System.TimeSpan.MaxValue">TimeSpan.MaxValue</see>.
```

File: xml/System.Threading/CancellationTokenSource.xml [hunk 5]
```diff
@@ -515,16 +516,16 @@
...
+## Remarks
+ The countdown for the delay starts during this call. When the delay expires, this <xref:System.Threading.CancellationTokenSource> is canceled, if it has not been canceled already.
 
+ Subsequent calls to `CancelAfter` will reset the delay for this <xref:System.Threading.CancellationTokenSource>, if it has not been canceled already.  
             ]]></format>
         </remarks>
         <exception cref="T:System.ObjectDisposedException">The exception thrown when this <see cref="T:System.Threading.CancellationTokenSource" /> has been disposed.</exception>
-        <exception cref="T:System.ArgumentOutOfRangeException">The exception that is thrown when <paramref name="delay" /> is less than -1 or greater than Int32.MaxValue.</exception>
+        <exception cref="T:System.ArgumentOutOfRangeException">
+          <paramref name="delay" />.<see cref="P:System.TimeSpan.TotalMilliseconds" /> is less than -1 or greater than Int32.<see cref="F:System.Int32.MaxValue" /> (or UInt32.<see cref="F:System.UInt32.MaxValue" />-1 on some versions of .NET). Note that this upper bound is more restrictive than TimeSpan.<see cref="F:System.TimeSpan.MaxValue" />.
```
Comment (gewarren) [inline xml/System.Threading/CancellationTokenSource.xml:527]:
```suggestion
          <paramref name="delay" />.<see cref="P:System.TimeSpan.TotalMilliseconds" /> is less than -1 or greater than <see cref="F:System.Int32.MaxValue">Int32.MaxValue</see> (or <see cref="F:System.UInt32.MaxValue">UInt32.MaxValue</see> - 1 on some versions of .NET). Note that this upper bound is more restrictive than <see cref="F:System.TimeSpan.MaxValue">TimeSpan.MaxValue</see>.
```

**Review by gewarren — COMMENTED** [review]
It doesn't look very good in the preview, and I got an idea about changing the link text. I will go ahead and commit these changes now.

https://review.docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtokensource.cancelafter?view=net-7.0&branch=pr-en-us-8036#system-threading-cancellationtokensource-cancelafter(system-timespan)

**Comment by gewarren** [comment]
@arunjvs It looks like `<see cref>` doesn't honor the link text you put in. I can see it used in a few other places in our repo and it's not honored there either. I actually think this is a bug, so I will log one.

**Comment by gewarren** [comment]
I've logged https://dev.azure.com/ceapex/Engineering/_workitems/edit/603788. Meanwhile, I'll convert these to see href links so we can see the full link text.

File: xml/System.Threading/CancellationTokenSource.xml [hunk 6]
```diff
@@ -259,7 +259,8 @@
  ]]></format>
         </remarks>
         <exception cref="T:System.ArgumentOutOfRangeException">
-          <paramref name="delay" />.<see cref="P:System.TimeSpan.TotalMilliseconds" /> is less than -1 or greater than <see cref="F:System.Int32.MaxValue" />.</exception>
+          <paramref name="delay" />.<see cref="P:System.TimeSpan.TotalMilliseconds" /> is less than -1 or greater than <see cref="F:System.Int32.MaxValue">Int32.MaxValue</see> (or <see cref="F:System.UInt32.MaxValue">UInt32.MaxValue</see> - 1 on some versions of .NET). Note that this upper bound is more restrictive than <see cref="F:System.TimeSpan.MaxValue">TimeSpan.MaxValue</see>.
```
Comment (gewarren) [inline xml/System.Threading/CancellationTokenSource.xml:262]:
```suggestion
          <paramref name="delay" />.<see cref="P:System.TimeSpan.TotalMilliseconds" /> is less than -1 or greater than <see href="https://docs.microsoft.com/dotnet/api/system.int32.maxvalue">Int32.MaxValue</see> (or <see href="https://docs.microsoft.com/dotnet/api/system.uint32.maxvalue">UInt32.MaxValue</see> - 1 on some versions of .NET). Note that this upper bound is more restrictive than <see href="https://docs.microsoft.com/dotnet/api/system.timespan.maxvalue">TimeSpan.MaxValue</see>.
```

File: xml/System.Threading/CancellationTokenSource.xml [hunk 7]
```diff
@@ -515,16 +516,16 @@
...
+ The countdown for the delay starts during this call. When the delay expires, this <xref:System.Threading.CancellationTokenSource> is canceled, if it has not been canceled already.
 
+ Subsequent calls to `CancelAfter` will reset the delay for this <xref:System.Threading.CancellationTokenSource>, if it has not been canceled already.  
             ]]></format>
         </remarks>
         <exception cref="T:System.ObjectDisposedException">The exception thrown when this <see cref="T:System.Threading.CancellationTokenSource" /> has been disposed.</exception>
-        <exception cref="T:System.ArgumentOutOfRangeException">The exception that is thrown when <paramref name="delay" /> is less than -1 or greater than Int32.MaxValue.</exception>
+        <exception cref="T:System.ArgumentOutOfRangeException">
+          <paramref name="delay" />.<see cref="P:System.TimeSpan.TotalMilliseconds" /> is less than -1 or greater than <see cref="F:System.Int32.MaxValue">Int32.MaxValue</see> (or <see cref="F:System.UInt32.MaxValue">UInt32.MaxValue</see> - 1 on some versions of .NET). Note that this upper bound is more restrictive than <see cref="F:System.TimeSpan.MaxValue">TimeSpan.MaxValue</see>.
```
Comment (gewarren) [inline xml/System.Threading/CancellationTokenSource.xml:527]:
```suggestion
          <paramref name="delay" />.<see cref="P:System.TimeSpan.TotalMilliseconds" /> is less than -1 or greater than <see href="https://docs.microsoft.com/dotnet/api/system.int32.maxvalue">Int32.MaxValue</see> (or <see href="https://docs.microsoft.com/dotnet/api/system.uint32.maxvalue">UInt32.MaxValue</see> - 1 on some versions of .NET). Note that this upper bound is more restrictive than <see href="https://docs.microsoft.com/dotnet/api/system.timespan.maxvalue">TimeSpan.MaxValue</see>.
```
